### DISCUSSION TITLE
Can't `cue eval` with an expression

### DISCUSSION THREAD

**Discussion by Winael** [cuelang/cue#1021] [Q&A] [answered]
Hi everyone,

as a new comer to cuelang, I'm trying create a personal side project to train myself. Basically is to create a dataset os Stadia games, that i could request with cue to create tools that will help me to create shortcuts, searching games from CLI, etc.
Nothing important, but it help me to understand how cuelang works, and be able then to share my knowledge around me to advocate around cuelang (and why not, this project could be useful for Stadia users too)

So this is my repo right know :

https://github.com/Winael/stadiaCue

The thing is, when i try to evaluate my code without any expression, it works

```bash
$ cue eval ./... && echo 'OK' || echo 'KO'
[...]
gameList: ["Crayta", "Assassin's Creed Odyssey", "Assassin's Creed Origins", "Assassin's Creed Syndicate", "Assassin’s Creed Unity", "Assassin's Creed Valhalla", "Attack On Titan 2: Final Battle", "AVICII Invector", "Baldur's Gate 3 Early Access", "Borderlands 3", "Cake Bash", "Celeste", "Chronos: Before the Ashes", "Cosmic Star Heroine", "Ary and the Secret of Seasons", "Cthulhu Saves Christmas", "Cyberpunk 2077", "Darksiders Genesis", "Dead by Daylight", "Destiny 2", "Destroy All Humans!", "DIRT 5", "Disco Elysium - The Final Cut", "Doom 2016", "DOOM 64", "DOOM Eternal Standard Edition", "DRAGON BALL XENOVERSE 2", "DRAGON QUEST XI S: Echoes of an Elusive Age", "A Place for the Unwilling"]
// ---
purchasedGames: ["assassins_creed_odyssey", "assassins_creed_origins", "assassins_creed_unity", "cake_bash", "crayta", "dragon_quest_xi_echoes_of_an_elusive_age"]
claimedGames: ["ary_and_the_secret_of_seasons", "avicii_invector", "celeste", "crayta", "cthulhu_saves_christmas", "dead_by_daylight", "destiny_2"]
OK
```

but if i try to evaluate with an expression, for ex to request the `gameList`

```bash
cue eval -e 'gameList' ./... && echo "OK" || echo "KO"
// /home/vjobard/Documents/Stadia/stadiaCue/channels
reference "gameList" not found:
    --expression:1:1
KO
```

And I can't export anything.

I'm pretty sure I miss something but I don't know what.

If someone could help me on this, it could be very nice.

Thx a lot
Winael

**Comment by myitcv** [comment, marked as answer]
@Winael after a quick glance, the reason this is happening is that `gameList` is not a field defined in all the `./...` packages. Instead it's only defined for `./games/all_games/...` packages:

```
$ cue eval -e gameList ./games/all_games/...
["Crayta", "Assassin's Creed Odyssey", "Assassin's Creed Origins", "Assassin's Creed Syndicate", "Assassin’s Creed Unity", "Assassin's Creed Valhalla", "Attack On Titan 2: Final Battle", "AVICII Invector", "Baldur's Gate 3 Early Access", "Borderlands 3", "Cake Bash", "Celeste", "Chronos: Before the Ashes", "Cosmic Star Heroine", "Ary and the Secret of Seasons", "Cthulhu Saves Christmas", "Cyberpunk 2077", "Darksiders Genesis", "Dead by Daylight", "Destiny 2", "Destroy All Humans!", "DIRT 5", "Disco Elysium - The Final Cut", "Doom 2016", "DOOM 64", "DOOM Eternal Standard Edition", "DRAGON BALL XENOVERSE 2", "DRAGON QUEST XI S: Echoes of an Elusive Age", "A Place for the Unwilling"]
```

**Reply by Winael** [reply]
OK, thx for your help

I believed that my package was `games`, so it's not totally clear

When I run 

```
cue eval -e gameList ./games/all_games/...
```

`./games/all_games/...` is in fact where I can find the leafs where my data are, isn't it ?

Regards

**Reply by myitcv** [reply]
> `./games/all_games/...` is in fact where I can find the leafs where my data are, isn't it ?

`./games/all_games/...` is simply a specification of the inputs to `cmd/cue`, packages in this case. See `cue help inputs` for more details: it explains the various inputs `cmd/cue` can consume, along with details of what `...` means etc.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/1021.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
